dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -139,4 +159,100 @@ public CollectionResource wrapResource(CollectionRest collection, String... rels
+                            + owningCommunityUuid + " not found");
+                }
+            }
+            collection = cs.create(context, parent);
+            cs.update(context, collection);
+            if (collectionRest.getMetadata() != null) {
+                for (MetadataEntryRest mer : collectionRest.getMetadata()) {
+                    String[] metadatakey = mer.getKey().split("\\.");
+                    cs.addMetadata(context, collection, metadatakey[0], metadatakey[1],
+                                   metadatakey.length == 3 ? metadatakey[2] : null, mer.getLanguage(), mer.getValue());
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+
+        return converter.convert(collection);
+    }
+
+
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected CollectionRest put(Context context, HttpServletRequest request, String apiCategory, String model, UUID id,
+                                JsonNode jsonNode)
+        throws RepositoryMethodNotImplementedException, SQLException, AuthorizeException {
+        CollectionRest collectionRest = null;
+        try {
+            collectionRest = new ObjectMapper().readValue(jsonNode.toString(), CollectionRest.class);
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+        }
+        Collection collection = cs.find(context, id);
+        if (collection == null) {
+            throw new ResourceNotFoundException(apiCategory + "." + model + " with id: " + id + " not found");
+        }
+        if (StringUtils.equals(id.toString(), collectionRest.getId())) {
+            List<MetadataEntryRest> metadataEntryRestList = collectionRest.getMetadata();
+            collection = (Collection) dspaceObjectUtils.replaceMetadataValues(context,
+                                                                              collection,
+                                                                              metadataEntryRestList);
+        } else {
+            throw new IllegalArgumentException("The UUID in the Json and the UUID in the url do not match: "
+                                                   + id + ", "
+                                                   + collectionRest.getId());
+        }
+        return converter.fromModel(collection);
+    }
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected void delete(Context context, UUID id) throws AuthorizeException {
+        Collection collection = null;
+        try {
+            collection = cs.find(context, id);
+            if (collection == null) {
+                throw new ResourceNotFoundException(
+                    CollectionRest.CATEGORY + "." + CollectionRest.NAME + " with id: " + id + " not found");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
More useful error message please

Reply:
Same as above, this could say something more descriptive like...
`throw new RuntimeException("Unable to delete Collection with id=" + id, e);`